ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# var.tfvars used to provision the powervs nodes is copied to the ${SHARED_DIR}
+echo "Invoking upi deprovision heterogeneous powervs"
+
+IBMCLOUD_HOME_FOLDER=/tmp/ibmcloud
+mkdir -p ${IBMCLOUD_HOME_FOLDER}
+REGION="${LEASED_RESOURCE}"
+
+if [ -z "$(command -v ibmcloud)" ]; then
+  echo "ibmcloud CLI doesn't exist, installing"
+  curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+fi
+
+function ic() {
+  HOME=${IBMCLOUD_HOME_FOLDER} ibmcloud "$@"
+}
+
+ic version
+ic login --apikey @${CLUSTER_PROFILE_DIR}/ibmcloud-api-key -r ${REGION}
+ic plugin install -f cloud-internet-services vpc-infrastructure cloud-object-storage power-iaas is
+
+echo "Check if the var.tfvars exists in ${SHARED_DIR}"
+if [ -f "${SHARED_DIR}/var.tfvars" ]
+then
+  # upgrade go to 1.20.7 or greater
+  if [ -z "$(command -v go)" ]; then
+    echo "go is not installed, proceed to installing go"
+    cd /tmp && wget -q https://go.dev/dl/go1.20.7.linux-amd64.tar.gz && tar -C /tmp -xzf go1.20.7.linux-amd64.tar.gz \
+      && export PATH=$PATH:/tmp/go/bin && export GOCACHE=/tmp/go && export GOPATH=/tmp/go && export GOMODCACHE="/tmp/go/pkg/mod"
+    if [ -z "$(command -v go)" ]; then
+      echo "Installed go successfully"
+    fi
+  fi
+
+  # build terraform from source v1.5.5
+  cd ${IBMCLOUD_HOME_FOLDER} && curl -L https://github.com/hashicorp/terraform/archive/refs/tags/v1.5.5.tar.gz -o ${IBMCLOUD_HOME_FOLDER}/terraform.tar.gz \
```

Comment:
Same thing with the embedded versioning here. Let's make it easy to update these in the future - be it in this patch or a follow up.

Reply:
Fix is in next commit.